dask/dataframe/io/tests/test_parquet.py

```diff
@@ -938,6 +938,112 @@ def test_to_parquet_default_writes_nulls(tmpdir):
     assert table[1].null_count == 2
 
 
+def test_to_parquet_pyarrow_w_inconsistent_schema_by_partition_fails_by_default(tmpdir):
+    check_pyarrow()
+
+    df = pd.DataFrame(
+        {"partition_column": [0, 0, 1, 1], "strings": ["a", "b", None, None]}
+    )
+
+    ddf = dd.from_pandas(df, npartitions=2)
+    ddf.to_parquet(str(tmpdir), engine="pyarrow", partition_on=["partition_column"])
+
+    # Test that read fails because of default behavior when schema not provided
+    with pytest.raises(ValueError) as e_info:
+        dd.read_parquet(str(tmpdir), engine="pyarrow", gather_statistics=False)
```

Comment:
You probably need to append `.compute()` here to actually raise the error.

Reply:
I don't believe so.

Reply:
Have you seen this test failing somewhere?

Reply:
Interesting, your right... that test doesn't seem to cause any issues in CI.  Hoever, it definitly fails locally in my branch without the `compute()`

Reply:
Definitely if we can identify failures without calling `compute`, that is preferable.  Users prefer getting errors as soon as possible.

Reply:
Right - It is best to get errors without calling compute.  However, my branch has the `pyarrow.read_metadata` changes which avoids "scanning" files for metadata when `gather_statistics=False`.  In this special case, `compute` needs to be called before problems with the schema in a particular partition can be detected.

Reply:
Locally the test works passes for me too. I wouldn't, before your PR, expect to need to call compute because `read_parquet` alone looks at metadata.